x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Sin.java

```diff
@@ -21,7 +23,10 @@
  * Sine trigonometric function.
  */
 public class Sin extends AbstractTrigonometricFunction {
-    public Sin(Source source, @Named("n") Expression n) {
+
+    @Described("Returns the trigonometric sine of an angle")
+    @Typed("double")
```

Comment:
The type information is already available in getDataType() - better to get it from there to avoid getting out of sync.

Reply:
`dataType()` method is not an option here unfortunately, it's an instance method and here we don't have function instances, but only classes and reflection.
Even worse, some functions return different types depending on the input, so calling that method makes sense only in the context of a query.